src/Xamarin.Android.Build.Tasks/Xamarin.Android.Tooling.targets

```diff
@@ -69,7 +74,7 @@ projects.
         JavaSdkPath="$(JavaSdkDirectory)"
         MinimumSupportedJavaVersion="$(MinimumSupportedJavaVersion)"
         LatestSupportedJavaVersion="$(LatestSupportedJavaVersion)"
-        ReferenceAssemblyPaths="$(_XATargetFrameworkDirectories)">
+        ReferenceAssemblyPaths="@(_AndroidApiInfoDirectories)">
```

Comment:
I have some concerns with moving the AndroidApiInfo.xml files away from Mono.Android.dll. I think this change would break framework assembly detection in certain tasks, but maybe I am missing something in the legacy vs .NET 6 target abstraction:
https://github.com/xamarin/xamarin-android/blob/ad5f6e85b09baf2644b6131667b9631b64ad0cf3/src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.Linker.cs#L56

Reply:
The problem was the old code:

https://github.com/xamarin/xamarin-android/blob/f1fe43bde20cd5ae0d8d02bfe0892bf02b029489/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Tooling.targets#L57-L63

It won't have the path here for the 31 and 32 packs at the same time, as you only end up with one.

Let me write a test for `MonoAndroidHelper.IsFrameworkAssembly()` and see what's happening there. This was "green", but it doesn't mean there wasn't some small bug here...

Reply:
I reviewed usage of `MonoAndroidHelper.TargetFrameworkDirectories`, and it appears to be OK:
* `<ResolveAssemblies/>` task is legacy only
* Some linker steps, but this is passed in a different way:

https://github.com/xamarin/xamarin-android/blob/8b17fbe46d3036136913b37054749c6bc613667e/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets#L46

I think this PR is OK, since it leaves `$(_XATargetFrameworkDirectories)` intact as it was before.

Something that might clean this up, would be if we built a .NET 6-only version of `Xamarin.Android.Build.Tasks.dll`? We could `#if` to remove some parts of the code that doesn't apply? But it would need to remain `netstandard2.0`, so it would work with .NET framework MSBuild. Something for a future PR?